service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/CbsReceiverLinkHandler.java

```diff
@@ -0,0 +1,121 @@
+        this.receiverLinkTag = RECEIVER_LINK_TAG_PREFIX;
+        this.receiverLinkAddress = RECEIVER_LINK_ENDPOINT_PATH;
+        this.authenticationMessageCallback = authenticationMessageCallback;
+    }
+
+    protected void setAuthenticationMessageCorrelationId(UUID authenticationMessageCorrelationId)
+    {
+        this.authenticationMessageCorrelationId = authenticationMessageCorrelationId;
+    }
+
+    static String getCbsTag()
+    {
+        return RECEIVER_LINK_TAG_PREFIX;
+    }
+
+    @Override
+    public String getLinkInstanceType()
+    {
+        return LINK_TYPE;
+    }
+
+    @Override
+    public void onDelivery(Event event)
+    {
+        Delivery delivery = event.getDelivery();
+        log.trace("Received a message on the CBS receiver link");
+        handleCBSResponseMessage();
+        delivery.free();
+    }
+
+    private void handleCBSResponseMessage()
+    {
+        AmqpsMessage amqpsMessage = super.getMessageFromReceiverLink();
+        if (amqpsMessage != null)
+        {
+            if (amqpsMessage.getApplicationProperties() != null && amqpsMessage.getProperties() != null)
+            {
+                Properties properties = amqpsMessage.getProperties();
+                UUID correlationId = (UUID) properties.getCorrelationId();
+                Map<String, Object> applicationProperties = amqpsMessage.getApplicationProperties().getValue();
+
+                if (!this.authenticationMessageCorrelationId.equals(correlationId))
+                {
+                    log.error("Received cbs authentication message with unexpected correlation id. Ignoring it...");
+                    amqpsMessage.acknowledge(Released.getInstance());
+                    return;
+                }
+
+                for (Map.Entry<String, Object> entry : applicationProperties.entrySet())
+                {
+                    String propertyKey = entry.getKey();
+                    if (propertyKey.equals(APPLICATION_PROPERTY_STATUS_CODE) && entry.getValue() instanceof Integer)
+                    {
+                        int authenticationResponseCode = (int) entry.getValue();
+
+                        String statusDescription = "";
+                        if (applicationProperties.containsKey(APPLICATION_PROPERTY_STATUS_DESCRIPTION))
+                        {
+                            statusDescription = (String) applicationProperties.get(APPLICATION_PROPERTY_STATUS_DESCRIPTION);
+                        }
+
+                        DeliveryState ackType = authenticationMessageCallback.handleAuthenticationResponseMessage(authenticationResponseCode, statusDescription);
+                        amqpsMessage.acknowledge(ackType);
+                        return;
+                    }
+                }
+            }
+            else
+            {
+                log.warn("Could not handle authentication message because it had no application properties or had no system properties");
```

Comment:
nit: No need to repeat 'had no' - Could not handle authentication message because it had no application properties or  system properties